starport/interface/cli/starport/cmd/network_chain_join.go

```diff
@@ -94,25 +96,40 @@ func networkChainJoinHandler(cmd *cobra.Command, args []string) error {
 	}
 
 	// ask to create an account on target blockchain.
+	shouldCreateAccount := true
+
 	printSection(fmt.Sprintf("Account on the blockchain %s", chainID))
-	account, err = createChainAccount(cmd.Context(), blockchain, fmt.Sprintf("%s blockchain", chainID), accountName)
-	if err != nil {
-		return err
+
+	if gentxPath != "" {
+		askAccountProposal := promptui.Prompt{
+			Label:     "Would you like to propose an account to place in Genesis",
+			IsConfirm: true,
+		}
+		_, err := askAccountProposal.Run()
+		shouldCreateAccount = err == nil
 	}
 
-	// ask to create an account proposal,
-	printSection("Account proposal")
+	if shouldCreateAccount {
+		acc, err := createChainAccount(cmd.Context(), blockchain, fmt.Sprintf("%s blockchain", chainID), accountName)
+		if err != nil {
+			return err
+		}
+		account = &acc
 
-	accQuestions := cliquiz.NewQuestion("Account coins",
-		&account.Coins,
-		cliquiz.DefaultAnswer(accountCoins),
-	)
+		// ask to create an account proposal.
+		printSection("Account proposal")
 
-	if err := cliquiz.Ask(accQuestions); err != nil {
-		return err
+		accQuestions := cliquiz.NewQuestion("Account coins",
```

Comment:
Do we want to make account proposal optional but in that case ask for coins when `-gentx` is not provided?

Reply:
If I understand correctly, if `--gentx` is not used, we follow the automatic path and always prompt for account to add to the genesis. Basically, like the way it's implemented right now.